```
Can not load RPM file: null/noarch/ceph-release-1-0.el9.noarch.rpm.
Error: No packages marked for reinstall.
Error: building at STEP "RUN true     && (. /etc/os-release ; if [ "$ID" = centos -a "$VERSION" = 8 ]; then find /etc/yum.repos.d/ -name '*.repo' -exec sed -i -e 's|^mirrorlist=|#mirrorlist=|g' -e 's|^#baseurl=http://mirror.centos.org/|baseurl=https://vault.centos.org/|g'  {} \; ; fi )     && if [ ! -f /etc/yum.repos.d/ceph.repo -a "$CEPH_IS_DEVEL" = true ]; then if [ -z "$CEPH_SHA1" ]; then CEPH_SHA1="$(sed -n 's/.*CEPH_GIT_VER *= *"\(.*\)".*/\1/p' /usr/bin/ceph)"; fi ; yum reinstall -y "$(curl -fs "https://shaman.ceph.com/api/search/?project=ceph&distros=centos/9/x86_64&flavor=default&ref=${CEPH_REF}&sha1=${CEPH_SHA1:-latest}" | jq -r .[0].url)/noarch/ceph-release-1-0.el9.noarch.rpm"; fi     && if [ ! -f /etc/yum.repos.d/ceph.repo -a "$CEPH_IS_DEVEL" != true ]; then yum reinstall -y "[https://download.ceph.com/rpm-${CEPH_REF}/el9/noarch/ceph-release-1-1.el9.noarch.rpm](https://download.ceph.com/rpm-$%7BCEPH_REF%7D/el9/noarch/ceph-release-1-1.el9.noarch.rpm)"; fi      && ( dnf config-manager --disable tcmu-runner,tcmu-runner-source,tcmu-runner-noarch,ceph-iscsi,ganesha || true )      && true": while running runtime: exit status 1
make: *** [Makefile:245: .test-container-id] Error 1
```

@Nikhil-Ladha, are you aware for this failure recently?

--- a/internal/rbd/group/util.go
+++ b/internal/rbd/group/util.go
@@ -434,8 +434,9 @@ func (cvg *commonVolumeGroup) GetCreationTime(ctx context.Context) (*time.Time, 
 // volumegroups should continue based on the type of error encountered.
 //
 // It checks if the given error matches any of the following known errors:
-//   - ErrPoolNotFound: The rbd pool where the volumegroup/omap is expected doesn't exist.
-//   - ErrGroupNotFound: The volumegroup doesn't exist in the rbd pool.
+//   - util.ErrPoolNotFound: The rbd pool where the volumegroup/omap is expected doesn't exist.
+//   - util.ErrConfigNotFound: No configuration exists for the cluster ID associated with the volumegroup.
+//   - rbderrors.ErrGroupNotFound: The volumegroup doesn't exist in the rbd pool.
 //   - rados.ErrPermissionDenied: Permissions to access the pool is denied.
 //
 // If any of these errors are encountered, the function returns `true`, indicating
@@ -451,6 +452,7 @@ func ShouldRetryVolumeGroupGeneration(err error) bool {
 	}
 	// Continue searching for specific known errors
 	return (errors.Is(err, util.ErrPoolNotFound) ||
+		errors.Is(err, util.ErrConfigNotFound) ||
 		errors.Is(err, rbderrors.ErrGroupNotFound) ||
 		errors.Is(err, rados.ErrPermissionDenied))
 }

--- a/internal/rbd/group/util_test.go
+++ b/internal/rbd/group/util_test.go
@@ -47,7 +47,12 @@ func Test_shouldRetryVolumeGroupGeneration(t *testing.T) {
 			want: true, // Known error, continue searching
 		},
 		{
-			name: "ErrGroupNotFound (continue searching)",
+			name: "ErrConfigNotFound (continue searching)",
+			args: args{err: util.ErrConfigNotFound},
+			want: true, // Known error, continue searching
+		},
+		{
+			name: "ErrRBDGroupNotFound (continue searching)",
 			args: args{err: rbderrors.ErrGroupNotFound},
 			want: true, // Known error, continue searching
 		},

--- a/internal/rbd/rbd_util.go
+++ b/internal/rbd/rbd_util.go
@@ -1260,6 +1260,7 @@ func generateVolumeFromVolumeID(
 // It checks if the given error matches any of the following known errors:
 //   - util.ErrKeyNotFound: The key required to locate the volume is missing in Rados omap.
 //   - util.ErrPoolNotFound: The rbd pool where the volume/omap is expected doesn't exist.
+//   - util.ErrConfigNotFound: No configuration exists for the cluster ID associated with the volume.
 //   - ErrImageNotFound: The image doesn't exist in the rbd pool.
 //   - rados.ErrPermissionDenied: Permissions to access the pool is denied.
 //
@@ -1277,6 +1278,7 @@ func ShouldRetryVolumeGeneration(err error) bool {
 	// Continue searching for specific known errors
 	return (errors.Is(err, util.ErrKeyNotFound) ||
 		errors.Is(err, util.ErrPoolNotFound) ||
+		errors.Is(err, util.ErrConfigNotFound) ||
 		errors.Is(err, rbderrors.ErrImageNotFound) ||
 		errors.Is(err, rados.ErrPermissionDenied))
 }

--- a/internal/rbd/rbd_util_test.go
+++ b/internal/rbd/rbd_util_test.go
@@ -415,6 +415,11 @@ func Test_shouldRetryVolumeGeneration(t *testing.T) {
 			want: true, // Known error, continue searching
 		},
 		{
+			name: "ErrConfigNotFound (continue searching)",
+			args: args{err: util.ErrConfigNotFound},
+			want: true, // Known error, continue searching
+		},
+		{
 			name: "ErrImageNotFound (continue searching)",
 			args: args{err: rbderrors.ErrImageNotFound},
 			want: true, // Known error, continue searching

--- a/internal/util/csiconfig.go
+++ b/internal/util/csiconfig.go
@@ -83,7 +83,7 @@ func readClusterInfo(pathToConfig, clusterID string) (*kubernetes.ClusterInfo, e
 		}
 	}
 
-	return nil, fmt.Errorf("missing configuration for cluster ID %q", clusterID)
+	return nil, fmt.Errorf("%w: %q", ErrConfigNotFound, clusterID)
 }
 
 // Mons returns a comma separated MON list from the csi config for the given clusterID.

--- a/internal/util/errors.go
+++ b/internal/util/errors.go
@@ -34,4 +34,6 @@ var (
 	ErrClusterIDNotSet = errors.New("clusterID must be set")
 	// ErrMissingConfigForMonitor is returned when clusterID is not found for the mon.
 	ErrMissingConfigForMonitor = errors.New("missing configuration of cluster ID for monitor")
+	// ErrConfigNotFound is returned when no configuration is found for a cluster ID.
+	ErrConfigNotFound = errors.New("missing configuration for cluster ID")
 )

